### PR TITLE
Add presubmit job for Cluster API Add-on Provider for Helm

### PR DESCRIPTION
**Pull request by Jont828**
Add presubmit job for [Cluster API Add-on Provider for Helm](https://github.com/kubernetes-sigs/cluster-api-addon-provider-helm) for PR validation.

### CODE REVIEW

**Comment by Jont828** [comment]
@dtzar @CecileRobertMichon I'm working on adding some presubmit jobs so we can run things like `make verify` and the linter for CAAPH. Is there anything else I need to configure besides creating this file?

**Comment by chases2** [comment]
Looks good to me overall. Looking for input on whether or not the Service Account name is correct, though (`prowjob-default-sa`)

/lgtm
/assign @listx

File: config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml [hunk 1]
```diff
@@ -0,0 +1,119 @@
...
+        - runner.sh
+        - ./scripts/ci-test.sh
+        env:
+        # This value determines the minimum Kubernetes
+        # supported version for Cluster API management cluster.
+        #
+        # To check the available envtest in Kubebuilder, please
+        # refer to https://github.com/kubernetes-sigs/kubebuilder/tree/tools-releases.
+        - name: KUBEBUILDER_ENVTEST_KUBERNETES_VERSION
+          value: "1.20.2"
```
Comment (dtzar) [inline config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml:113]: Does CAPI really support 1.20.2 management cluster even though it is so old?

Comment (Jont828) [reply config/jobs/kubernetes-sigs/cluster-api-addon-provider-helm/cluster-api-addon-provider-helm-presubmits-main.yaml:113]: That's a good question. I copied this file from the [CAPI presubmit jobs](https://github.com/kubernetes/test-infra/blob/358eb42ed2d3085e3f67d898f3ab52b7a30989c6/config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-main.yaml#L90) so it seems like it is running 1.20.

**Comment by Jont828** [comment]
@chases2 I copied the presubmit file from the [CAPI presubmit file](https://github.com/kubernetes/test-infra/blob/358eb42ed2d3085e3f67d898f3ab52b7a30989c6/config/jobs/kubernetes-sigs/cluster-api/cluster-api-presubmits-main.yaml#L90) which references that account.

One question I did have was about verifying the configuration. I looked at [these docs](https://github.com/kubernetes/test-infra/tree/master/config/testgrids) and when I ran `go test ./config/tests` from the root of the repo, I simply got the following error. Am I doing this correctly?
```
no Go files in /home/jonathan/go/src/test-infra/config/tests
```

**Review by jackfrancis — APPROVED** [review]
lgtm

as a first pass and starting the process of running testgrid jobs

**Comment by Jont828** [comment]
@CecileRobertMichon @jackfrancis @chases2 Are we good to merge this?

**Comment by listx** [comment]
Sorry this slipped my inbox. 

/approve
/lgtm

**Pull request merged**